src/overload_gradient.jl

```diff
@@ -1,67 +1,149 @@
+    end
 
-for fn in union(ops_1_to_2_sz, ops_1_to_2_fz)
-    @eval Base.$fn(t::T) where {T<:GlobalGradientTracer} = (t, empty(T))
+    @eval function Base.$fn(::Number, ty::GradientTracer)
+        return gradient_tracer_2_to_1_one_tracer(ty, is_firstder_arg2_zero_global($fn))
+    end
+    @eval function Base.$fn(x::Number, dy::D) where {P,T<:GradientTracer,D<:Dual{P,T}}
+        y = primal(dy)
+        p_out = Base.$fn(x, y)
+        t_out = gradient_tracer_2_to_1_one_tracer(
+            tracer(dy), is_firstder_arg2_zero_local($fn, x, y)
+        )
+        return Dual(p_out, t_out)
+    end
 end
 
-for fn in union(ops_1_to_2_zs, ops_1_to_2_zf)
-    @eval Base.$fn(t::T) where {T<:GlobalGradientTracer} = (empty(T), t)
+## 1-to-2
+function gradient_tracer_1_to_2(
+    t::T, is_firstder_out1_zero::Bool, is_firstder_out2_zero::Bool
+) where {T<:GradientTracer}
+    t1 = gradient_tracer_1_to_1(t, is_firstder_out1_zero)
+    t2 = gradient_tracer_1_to_1(t, is_firstder_out2_zero)
+    return (t1, t2)
 end
-for fn in ops_1_to_2_zz
-    @eval Base.$fn(::T) where {T<:GlobalGradientTracer} = (empty(T), empty(T))
+
+for fn in ops_1_to_2
+    @eval function Base.$fn(t::GradientTracer)
+        return gradient_tracer_1_to_2(
+            t, is_firstder_out1_zero_global($fn), is_firstder_out2_zero_global($fn)
+        )
+    end
+
+    @eval function Base.$fn(d::D) where {P,T<:GradientTracer,D<:Dual{P,T}}
+        x = primal(d)
+        p1_out, p2_out = Base.$fn(x)
+        t1_out, t2_out = gradient_tracer_1_to_2(
+            t, is_firstder_out1_zero_local($fn, x), is_firstder_out2_zero_local($fn, x)
+        )
+        return (Dual(p1_out, t1_out), Dual(p1_out, t1_out))
+    end
 end
 
-# Extra types required for exponent
-for T in (:Real, :Integer, :Rational)
-    @eval Base.:^(t::GlobalGradientTracer, ::$T) = t
-    @eval Base.:^(::$T, t::GlobalGradientTracer) = t
+# Extra types required for exponent 
+for S in (Real, Integer, Rational, Irrational{:ℯ})
+    Base.:^(t::GradientTracer, ::S) = t
+    Base.:^(::S, t::GradientTracer) = t
+
+    function Base.:^(dx::D, y::S) where {P,T<:GradientTracer,D<:Dual{P,T}}
+        return Dual(primal(dx)^y, tracer(dx))
+    end
+    function Base.:^(x::S, dy::D) where {P,T<:GradientTracer,D<:Dual{P,T}}
+        return Dual(x^primal(dy), tracer(dy))
+    end
 end
-Base.:^(t::GlobalGradientTracer, ::Irrational{:ℯ}) = t
-Base.:^(::Irrational{:ℯ}, t::GlobalGradientTracer) = t
 
 ## Rounding
-Base.round(t::T, ::RoundingMode; kwargs...) where {T<:GlobalGradientTracer} = empty(T)
+Base.round(t::T, ::RoundingMode; kwargs...) where {T<:GradientTracer} = empty(T)
+function Base.round(
+    d::D, mode::RoundingMode; kwargs...
+) where {P,T<:GradientTracer,D<:Dual{P,T}}
+    return Dual(round(primal(d), mode; kwargs...), empty(T))
+end
 
-## Random numbers
-rand(::AbstractRNG, ::SamplerType{T}) where {T<:GlobalGradientTracer} = empty(T)
+## Random numbers 
+# TODO: support random numbers on Duals
```

Comment:
what? how? why?

Reply:
For global tracers, this was trivial zu support: just return an empty tracer.
For Duals, I agree with dropping it.